core/src/network/reducers/get_links_count.rs

```diff
@@ -0,0 +1,130 @@
+use crate::{
+    action::{ActionWrapper, GetLinksKey, GetLinksKeyByTag},
+    network::{query::NetworkQuery, reducers::send, state::NetworkState},
+    state::State,
+};
+
+use holochain_core_types::{crud_status::CrudStatus, error::HolochainError};
+use holochain_json_api::json::JsonString;
+use holochain_net::connection::json_protocol::{JsonProtocol, QueryEntryData};
+use holochain_persistence_api::hash::HashString;
+
+fn reduce_get_links_inner(
+    network_state: &mut NetworkState,
+    key: &GetLinksKey,
+    crud_status: Option<CrudStatus>,
+) -> Result<(), HolochainError> {
+    network_state.initialized()?;
+    let query_json: JsonString =
+        NetworkQuery::GetLinksCount(key.link_type.clone(), key.tag.clone(), crud_status).into();
+    send(
+        network_state,
+        JsonProtocol::QueryEntry(QueryEntryData {
+            requester_agent_id: network_state.agent_id.clone().unwrap().into(),
+            request_id: key.id.clone(),
+            dna_address: network_state.dna_address.clone().unwrap(),
+            entry_address: HashString::from(key.base_address.clone()),
+            query: query_json.to_string().into_bytes(),
+        }),
+    )
+}
+
+fn reduce_get_links_inner_by_tag(
+    network_state: &mut NetworkState,
+    key: &GetLinksKeyByTag,
+    crud_status: Option<CrudStatus>,
+) -> Result<(), HolochainError> {
+    network_state.initialized()?;
+    let query_json: JsonString = NetworkQuery::GetLinksByTag(key.tag.clone(), crud_status).into();
+    send(
+        network_state,
+        JsonProtocol::QueryEntry(QueryEntryData {
+            requester_agent_id: network_state.agent_id.clone().unwrap().into(),
+            request_id: key.id.clone(),
+            dna_address: network_state.dna_address.clone().unwrap(),
+            entry_address: HashString::from(String::default()),
```

Comment:
Not sure about the ramifications of returning an empty hash @lucksus

Reply:
It won't work in non-full-sync scenario.
I didn't know where to put it, so I'll start here: **you are missing the link base everywhere.** That is what has to go in here!